src/js/jsx/sections/libraries/LibraryBar.jsx

```diff
@@ -63,53 +69,130 @@ define(function (require, exports, module) {
-            // FIXME: We're going to need context sensitive boxes for (stroke, fill and overlay color)
-            // For demonstration purposes, this action uses a hard coded color
+        addColorAsset: function (color) {
             // FIXME: We may also need to extend to other color spaces/representations here, check other uses of colors
-            this.getFlux().actions.libraries.createColorAsset({ r: 0, g: 255, b: 128 });
+            this.getFlux().actions.libraries.createColorAsset(color);
+        },
+        
+        /**
+         * Helper function to create color button based on the layer attribute (fills or strokes)
+         * @private
+         */
+        _createColorButton: function (attr, buttonText) {
+            var layer = this._selectedLayers().first();
+            
+            if (this._selectedLayers().size !== 1 || layer[attr].isEmpty()) {
+                return null;
+            }
+            
+            var color = layer[attr].first().color;
+            
+            return (
+                <SplitButtonItem title={buttonText} disabled={this.props.disabled}>
+                    <div className="split-button__item__color-icon"
+                         style={{ backgroundColor: color.toCssRGB() }}
+                         onClick={this.addColorAsset.bind(this, color)} />
+                </SplitButtonItem>
+            );
+        },
+        
+        /**
+         * True if user can add the selected layer(s) as graphic into the libraries.
+         * @private
+         */
+        _canAddGraphic: function () {
+            if (this.props.disabled || this._selectedLayers().isEmpty()) {
+                return false;
+            }
+            
+            var countOfEmptyGroups = 0;
+            
+            this._selectedLayers().forEach(function (layer) {
+                if (layer.kind === layer.layerKinds.GROUP &&
+                    this.props.document.layers.isEmptyGroup(layer)) {
+                    countOfEmptyGroups++;
+                }
+            }.bind(this));
+        
+            return countOfEmptyGroups !== this._selectedLayers().size;
+        },
+        
+        /**
+         * True if user can add the selected text layer into the libraries.
+         * @private
+         */
+        _canAddCharacterStyle: function () {
+            if (this.props.disabled ||
+                this._selectedLayers().size !== 1 ||
+                !this._selectedLayers().first().isTextLayer()) {
+                return false;
+            }
+            
+            var fontStore = this.getFlux().store("font"),
+                hasMissingFont = false;
+                
+            this._selectedLayers().first().text.characterStyles.forEach(function (style) {
+                if (!fontStore.getFontFamilyFromPostScriptName(style.postScriptName)) {
+                    hasMissingFont = true;
+                }
+            });
+            
+            return !hasMissingFont;
```

Comment:
If you're just going to return the flag, you can probably voodoo this with `.some()`.

However, since you have a check to see if selection size is 1, and we only use the FIRST character style in the layer, you should assign the first layer to a variable and just check it's `.text.characterStyles[0].postScriptName`